includes/shortcodes/class.llms.shortcode.checkout.php

```diff
@@ -114,6 +114,26 @@ private static function confirm_payment( $atts ) {
 
 	}
 
+	/**
+	 * Loads the 'product/not-purchasable.php' template and returns true if it printed an error notice,
+	 * else returns false.
+	 *
+	 * The default LifterLMS 'product/not-purchasable.php' template prints error notices if a related course
+	 * is restricted, e.g. if enrollment capacity has been reached or we're not in the enrollment period.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Product $product The product object.
+	 * @return bool
+	 */
+	private static function does_product_not_purchasable_template_print_error_notices( $product ) {
```

Comment:
This is `ingenious` although I must say that I don't really like it ehehe.

I see it's to reduce redundancy, but loading a "front" template, where accidentally there's some business logic we need here, doesn't seem the right solution to me, not only for a formal matter, but because templates are can be overridden by themes and 3rd party plugins, they're not that reliable.
Can't we just move the checks we need into functions/methods and use them in both places (here and the template)?

Reply:
This might be what we're looking for @pondermatic: https://developer.lifterlms.com/reference/functions/llms_notice_count/